controllers/metallb_controller.go

```diff
@@ -67,12 +68,30 @@ func (r *MetallbReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		// Error reading the object - requeue the request.
 		return ctrl.Result{}, err
 	}
+	err = status.Update(context.TODO(), r.Client, instance, status.ConditionProgressing, "", "")
+	if err != nil {
+		logger.Info("Failed to update metallb status", "Desired status", status.ConditionProgressing)
+	}
 
 	err = r.syncMetalLBResources(instance)
+	if err != nil {
+		err2 := status.Update(context.TODO(), r.Client, instance, status.ConditionDegraded, "FailedToSyncMetalLBResources", err.Error())
```

Comment:
I _think_ if you do if err := xxx; err {
}
the scope of the err will be local to the if and you can get rid of the ugly err2

Reply:
done